### PR TITLE
Unit Test for java.util.stream

### PR DESCRIPTION
**Pull request by gaurangkudale**
I've created a simple unit test for `java.util.stream` in this path `src\tests\StreamTest`

### CODE REVIEW

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho,
 As you said I've changed the code where the print statement is removed and I have got a few questions

1. Should I write the test cases for all of the methods present in `java.util.stream`?
2. As I've written the unit test for `IntStream` is it working fine? and is it mergeable to the main jpf-core

**Comment by gaurangkudale** [comment]
Hi@cyrille-artho 
Can you please solve my doubts so that I can keep contributing

**Comment by cyrille-artho** [comment]
The version I can see right now prints all the numbers. Typically, in a unit test, we want to avoid side effects such as printing on the console, and instead verify as much as possible using assertions.
Try to rewrite your code such that it uses `assertEquals` from JUnit.
To answer your first question, it is not necessary now to test all methods of Java Streams. If we test one or two methods, we can see that streams are supported as such in JPF. If a particular functionality does not seem to work well, we can create more unit tests later.

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho 
I've updated the code such that it is not printing the number in the console

**Comment by gaurangkudale** [comment]
Please check the `src\tests\StreamTest` folder to verify

**Comment by cyrille-artho** [comment]
Hi,
Thank you for the update. The challenge in the test is to ensure that `doSomething` is called in the right order by the stream. Each number from 0 to 9 should appear once in succession. Probably it's necessary to create a new (inner) class with a field that remembers the value of the counter, so it can count independently from 0 to 9 and verify that the call sequence is correct. Having a redundant counter allows you to encode the property in an assertion:
`assertEquals(value == expected);`
This assumes that `doSomething` is called with a parameter (`value`) and that `expected` is the name of the counter that goes from 0 to 9.
Also, please note the typo in `doSomething` at the moment, which you can fix alongside the suggested rewrite.

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho Currently I've started learning JUnit for testing so can you please tell me what exactly I should do with `doSomething`?

**Comment by cyrille-artho** [comment]
A useful unit test has at least one assertion (such as `assertEquals`) that checks if the value computed by a given function matches the expected value. In your example, the expected values for each call are a series from 0 to 9.
You therefore need to create a small class that allows you to instantiate an object that counts every time `doSomething` is called. Inside `doSomething`, you use your own counter and compare it to the value returned by the stream.
This way, you can confirm that the stream returns the numbers in the same order (if it is a sequential stream).

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho Is there something wrong about Unit Test

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho 
I've done some changes in the code but it is running fine on my machine why it is not passing the test cases over there any specific reason behind that?
Link of screenShots 1 :-  [https://kjeieduin-my.sharepoint.com/:i:/g/personal/14seit_tcoer_kjei_edu_in/EfzaXDUtyIpCsBz0xt5Mg-EBizf7AwOg9Y4XvBhGhlx8jw?e=f0KjoR](url)

Link 2  [https://kjeieduin-my.sharepoint.com/:i:/g/personal/14seit_tcoer_kjei_edu_in/EXQDEArP2cVLi7a-rY5WbDABXQ3-VhVm1L9YTjV3Lg06og?e=0cNmnt](url)

**Comment by gaurangkudale** [comment]
Is there anything I can do?

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho 
Can you check whether it is working on your system?

**Comment by cyrille-artho** [comment]
It looks like you import the assertions from the wrong package. Look at some existing tests to see which packages to import. Maybe it will suffice if you extend class `TestJPF`, as that is necessary for tests that run inside JPF; methods like `assert` are inherited then.

Also, move the test classes to their own package, e.g., to, `src/tests/java8` as package `java8`, to avoid cluttering the top-level directory with test files.

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho Did the changes as you suggested now can you please review the code and please let me know if  there are any changes

**Comment by gaurangkudale** [comment]
HI @cyrille-artho I think the changes are working fine can you please review the code

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho  I think the changes are working fine can you please review the code

**Comment by cyrille-artho** [comment]
Thanks for the update. I see that the code now compiles and runs, and the test passes.
However, please check the content of the test. It may be better to have something more simple such as "isEven" so we don't have to worry about the recursion. Most importantly, the assertion should check something that the stream computation returns. `assertEquals(2,2);` will always succeed no matter what.

File: src/classes/java/lang/ThreadLocal.java [hunk 1]
```diff
@@ -105,15 +105,13 @@ public void remove(){
     class threadSafeFormatter{
       public static ThreadLocal<SimpleDateFormat> df = ThreadLocal.withInitial(() 
           -> new SimpleDateFormat("yyyy-MM-dd"));
-  }
-  
   public class App  {
       public static void main(String[] args) throws Exception {
           threadSafeFormatter tf = new threadSafeFormatter();
           Thread t1 = new Thread();
           t1.start();
```
Comment (ramawatraksha75) [inline src/classes/java/lang/ThreadLocal.java]: tf.SimpleDateFormat();

Comment (gaurangkudale) [reply src/classes/java/lang/ThreadLocal.java]: Hi @ramawatraksha75  Can you please explain a bit more whats the issue is?

Comment (ramawatraksha75) [reply src/classes/java/lang/ThreadLocal.java]: Sure

Comment (ramawatraksha75) [reply src/classes/java/lang/ThreadLocal.java]: As I can see we have created the method for treadSafeFormatter but didn't called the function present  in that class

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho 
Can you please tell me more I didnt get that

File: src/classes/java/lang/ThreadLocal.java (hunk 1 above)
Comment (gaurangkudale) [reply src/classes/java/lang/ThreadLocal.java]: I guess it is not necessary

Comment (ramawatraksha75) [reply src/classes/java/lang/ThreadLocal.java]: Can you please help me out with that

Comment (cyrille-artho) [reply src/classes/java/lang/ThreadLocal.java]: This test is maybe not very strong in that it only checks the existence of the alternative constructor, but it is sufficient to check that the old model class of JPF did not support it and the new one does. So there is no need right now to change it.

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho
I was busy with academy work so that I didn't get the time for this issue now I'll continue to contribute

**Comment by cyrille-artho** [comment]
Hi Guarang,
Thank you for the update. You need to add a few more `import` statements for the code to compile, as the classes `Collectors` and `List` are not in the default package. So you need to add `import java.util.List` etc., until the code compiles and passes the tests.

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho 
As you said I've done some changes and tests are also passed please note that waiting for your reply 
thank you

**Comment by cyrille-artho** [comment]
Great, this unit test passes and is functional, so I will merge this PR and close #302.

**Comment by gaurangkudale** [comment]
Excited to Do more contribution 
@cyrille-artho  What should I do next?

**Pull request merged**